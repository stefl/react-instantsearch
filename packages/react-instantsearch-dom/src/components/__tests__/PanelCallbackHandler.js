import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PanelCallbackHandler from '../PanelCallbackHandler';

Enzyme.configure({ adapter: new Adapter() });

describe('PanelCallbackHandler', () => {
  it('expect to render', () => {
    const props = {
      canRefine: true,
    };

    const wrapper = shallow(
      <PanelCallbackHandler {...props}>
        <div>Hello content</div>
      </PanelCallbackHandler>
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('willMount', () => {
    it('expect to call setCanRefine when the context is given', () => {
      const props = {
        canRefine: true,
      };

      const context = {
        setCanRefine: jest.fn(),
      };

      shallow(
        <PanelCallbackHandler {...props}>
          <div>Hello content</div>
        </PanelCallbackHandler>,
        { context }
      );

      expect(context.setCanRefine).toHaveBeenCalledTimes(1);
      expect(context.setCanRefine).toHaveBeenCalledWith(true);
    });

    it('expect to not throw when the context is not given', () => {
      const props = {
        canRefine: true,
      };

      const trigger = () =>
        shallow(
          <PanelCallbackHandler {...props}>
            <div>Hello content</div>
          </PanelCallbackHandler>
        );

      expect(() => trigger()).not.toThrow();
    });
  });

  describe('willReceiveProps', () => {
    it('expect to call setCanRefine when the context is given', () => {
      const props = {
        canRefine: true,
      };

      const context = {
        setCanRefine: jest.fn(),
      };

      const wrapper = shallow(
        <PanelCallbackHandler {...props}>
          <div>Hello content</div>
        </PanelCallbackHandler>,
        { context }
      );

      wrapper.setProps({ canRefine: false });

      expect(context.setCanRefine).toHaveBeenCalledTimes(2);
      expect(context.setCanRefine).toHaveBeenLastCalledWith(false);
    });

    it('expect to not call setCanRefine when the nextProps is the same', () => {
      const props = {
        canRefine: true,
      };

      const context = {
        setCanRefine: jest.fn(),
      };

      const wrapper = shallow(
        <PanelCallbackHandler {...props}>
          <div>Hello content</div>
        </PanelCallbackHandler>,
        { context }
      );

      wrapper.setProps({ canRefine: true });

      expect(context.setCanRefine).toHaveBeenCalledTimes(1);
    });

    it('expect to not throw when the context is not given', () => {
      const props = {
        canRefine: true,
      };

      const trigger = () => {
        const wrapper = shallow(
          <PanelCallbackHandler {...props}>
            <div>Hello content</div>
          </PanelCallbackHandler>
        );

        wrapper.setProps({ canRefine: false });
      };

      expect(() => trigger()).not.toThrow();
    });
  });
});
