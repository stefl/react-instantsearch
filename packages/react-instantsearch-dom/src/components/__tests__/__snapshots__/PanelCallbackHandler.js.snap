// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelCallbackHandler expect to render 1`] = `
<div>
  Hello content
</div>
`;
